File: frontend/src/utils/tenant.js
export const TENANT_ID_PATTERN = /^[a-z0-9][a-z0-9_-]{1,63}$/;

export const TENANT_VALIDATION_HINT =
  "Usa 2 a 64 caracteres: letras minúsculas, números, guion (-) o guion bajo (_).";

export const sanitizeTenantId = (value, fallback = "public") => {
  const normalized = String(value || "")
    .trim()
    .toLowerCase();
  return normalized || fallback;
};

export const isValidTenantId = (value) => TENANT_ID_PATTERN.test(sanitizeTenantId(value));

File: backend/app/api/tenant.py
import re

from fastapi import Header, HTTPException, status

from app.core.config import settings

TENANT_HEADER_NAME = "X-Tenant-ID"
TENANT_ID_PATTERN = re.compile(r"^[a-z0-9][a-z0-9_-]{1,63}$")


def normalize_tenant_id(raw_tenant_id: str | None) -> str:
    tenant_id = (raw_tenant_id or settings.DEFAULT_TENANT_ID).strip().lower()
    if not TENANT_ID_PATTERN.fullmatch(tenant_id):
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=(
                "Tenant inválido. Use 2-64 caracteres con letras minúsculas, "
                "números, guion (-) o guion bajo (_)."
            ),
        )
    return tenant_id


def get_tenant_id(
    x_tenant_id: str | None = Header(default=None, alias=TENANT_HEADER_NAME),
) -> str:
    return normalize_tenant_id(x_tenant_id)

